.gitattributes

```diff
@@ -1,63 +1,54 @@
-###############################################################################
-# Set default behavior to automatically normalize line endings.
-###############################################################################
+# Auto detect text files and perform LF normalization
+# http://davidlaing.com/2012/09/19/customise-your-gitattributes-to-become-a-git-ninja/
 * text=auto encoding=UTF-8
 
-###############################################################################
-# Set default behavior for command prompt diff.
-#
-# This is need for earlier builds of msysgit that does not have it on by
-# default for csharp files.
-# Note: This is only used by command line
-###############################################################################
-#*.cs     diff=csharp
+#
+# The above will handle all files NOT found below
+#
+
+#
+## Binary files should be left untouched
+#
+*.bmp       binary
+*.gif       binary
+*.jpg       binary
+*.png       binary
+*.class     binary
+*.dll       binary
+*.ico       binary
+*.pfx       binary
+*.snk       binary
+.exe        binary
+*.pdb       binary
+
+#
+## Visual Studio
+#
+*.sln       text eol=crlf merge=union
+*.csproj    text eol=crlf merge=union
+*.vbproj    text eol=crlf merge=union
+
+#
+## These files are text and should be normalized (Convert crlf => lf)
+#
 
-###############################################################################
-# Set the merge driver for project and solution files
-#
-# Merging from the command prompt will add diff markers to the files if there
-# are conflicts (Merging from VS is not affected by the settings below, in VS
-# the diff markers are never inserted). Diff markers may cause the following 
-# file extensions to fail to load in VS. An alternative would be to treat
-# these files as binary and thus will always conflict and require user
-# intervention with every merge. To do so, just uncomment the entries below
-###############################################################################
-#*.sln       merge=binary
-#*.csproj    merge=binary
-#*.vbproj    merge=binary
-#*.vcxproj   merge=binary
-#*.vcproj    merge=binary
-#*.dbproj    merge=binary
-#*.fsproj    merge=binary
-#*.lsproj    merge=binary
-#*.wixproj   merge=binary
-#*.modelproj merge=binary
-#*.sqlproj   merge=binary
-#*.wwaproj   merge=binary
+# source code
+*.cs        text diff=csharp
+*.vb        text
+*.java      text diff=java
+*.resx      text merge=union
+*.html      text diff=html
+*.props     text eol=crlf
+*.xaml      text
+*.config    text
 
-###############################################################################
-# behavior for image files
-#
-# image files are treated as binary by default.
-###############################################################################
-#*.jpg   binary
-#*.png   binary
-#*.gif   binary
+# scripts
+*.sh        text eol=lf
+*.bat       text eol=crlf
+*.cmd       text eol=crlf
+*.ps1       text eol=crlf
```

Comment:
LF works on Windows, so that should probably be LF.

Reply:
There were comments saying there are some bugs with LF so for now I will keep it as CRLF.

Reply:
Good to know. Do you have a link?